gas/config/tc-arc.c

```diff
@@ -3820,6 +3824,11 @@ assemble_insn (const struct arc_opcode *opcode,
 				  pflags, nflg,
 				  operand->default_reloc);
 	      break;
+	    case O_jli:
+	      reloc = find_reloc ("jli", opcode->name,
```

Comment:
find_reloc is more complex, for example you may use @jli for various instructions, you need to generate for each instruction a specific relocation. This selection which relocation for which instruction can be done/added by find_reloc. In ur case we have a single reloc as indicated in the ARC_RELOC_TABLE.